src/main/java/org/openstreetmap/atlas/checks/validation/tag/MixedCaseNameCheck.java

```diff
@@ -0,0 +1,263 @@
+            "Mhic", "Mic");
+    private static final List<String> MIXED_CASE_UNITS_DEFAULT = Arrays.asList("kV");
+
+    // A list of countries where the name tag should be checked
+    private final List<String> checkNameCountries;
+    // A list of language specific name tags to check
+    private final List<String> languageNameTags;
+    // A list of prepositions that are normally lower case in names
+    private final List<String> lowerCasePrepositions;
+    // A list of articles that are normally lower case in names, unless at the start
+    private final List<String> lowerCaseArticles;
+    // A string of characters that can proceed a capital letter
+    private final String splitCharacters;
+    // A list of name affixes that can proceed a capital letter
+    private final String nameAffixes;
+    // Know intentionally mixed case words
+    private final String mixedCaseUnits;
+
+    /**
+     * The default constructor that must be supplied. The Atlas Checks framework will generate the
+     * checks with this constructor, supplying a configuration that can be used to adjust any
+     * parameters that the check uses during operation.
+     *
+     * @param configuration
+     *            the JSON configuration for this check
+     */
+    public MixedCaseNameCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.checkNameCountries = (List<String>) configurationValue(configuration,
+                "check_name.countries", CHECK_NAME_COUNTRIES_DEFAULT);
+        this.languageNameTags = (List<String>) configurationValue(configuration,
+                "name.language.keys", LANGUAGE_NAME_TAGS_DEFAULT);
+        this.lowerCasePrepositions = (List<String>) configurationValue(configuration,
+                "name.prepositions", LOWER_CASE_PREPOSITIONS_DEFAULT);
+        this.lowerCaseArticles = (List<String>) configurationValue(configuration, "name.articles",
+                LOWER_CASE_ARTICLES_DEFAULT);
+        this.splitCharacters = (String) configurationValue(configuration, "regex.split",
+                SPLIT_CHARACTERS_DEFAULT);
+        this.nameAffixes = (String) configurationValue(configuration, "name.affixes",
+                NAME_AFFIXES_DEFAULT, value -> String.join("|", (List<String>) value));
+        this.mixedCaseUnits = (String) configurationValue(configuration, "name.units",
+                MIXED_CASE_UNITS_DEFAULT, value -> String.join("|", (List<String>) value));
+    }
+
+    /**
+     * This function will validate if the supplied atlas object is valid for the check.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        // Valid objects are items that were OSM nodes or ways
```

Comment:
I would also add something at the end like "(Equivalent to Atlas nodes, points, edges, lines and areas)"